### ISSUE TITLE
Unused import not detected F401

### ISSUE THREAD

**Issue by schlamar** [astral-sh/ruff#15057]
```python
import mylib.mylib as cli
import mylib.test

cli.test()
```

Expected error:

```
test2.py:2:8: F401 [*] `mylib.test` imported but unused
```

Actual: 

```
> ruff check --select F .\test2.py
All checks passed!
> ruff --version
ruff 0.8.3
```

**Comment by MichaReiser** [comment]
See https://github.com/astral-sh/ruff/issues/4656

**Comment by MichaReiser** [comment]
As far as I can tell it's the same underlying issue and it should be fixed by https://github.com/astral-sh/ruff/pull/13965
